### PR TITLE
make test suite opt in

### PR DESCRIPTION
**Pull request by tjhei**
With this change, only a single test is added to the list of tests that
are going to be executed when you run "ctest" or "make test". To enable
all tests, either execute "make setup_tests" or configure with -D
ASPECT_RUN_ALL_TESTS=ON.

The reason for this change is, that many tests fail due to round-off
etc.. This confuses new users who expect "make; make test" to succeed.

### CODE REVIEW

**Comment by tjhei** [comment]
I am posting this to put it up for discussion. We should first check that it is working and decide on a small subset of tests (or write new tests) that hopefully succeed on most machines. Hello MACs?

Maybe a test that runs with MPI is a good idea? Should we write separate tests that do not check statistics? (we likely don't have numdiff after all)

**Comment by tjhei** [comment]
This whole thing was suggested by @gassmoeller .

**Comment by bangerth** [comment]
How about running only a single (maybe special) test that simply prints essentially your PR message above: "tests are not enabled by default because they typically differ by round off and some will fail on each platform even if everything is ok; you can run tests by saying xxx if you are sure that you understand the limitations of the testsuite and want to run tests anyway".

Is there a section on the testsuite in the manual that needs to be updated?

File: CMakeLists.txt [hunk 1]
```diff
@@ -108,6 +108,14 @@ CONFIGURE_FILE(
 
 
 # Next, set up the testsuite
+
+SET(ASPECT_RUN_ALL_TESTS OFF CACHE BOOL "Set up complete test suite to run.")
+
+ADD_CUSTOM_TARGET(setup_tests
+        COMMAND echo "Enabling all tests ..."
+	COMMAND ${CMAKE_COMMAND} -D ASPECT_RUN_ALL_TESTS=ON .
```
Comment (bangerth) [inline CMakeLists.txt]: space/tab issue?

**Comment by tjhei** [comment]
> How about running only a single (maybe special) test that simply prints essentially your PR message 

While printing a message sounds like a good idea, ctest by default won't show test output if the test succeeds. So I don't know how to print this message. The only thing I can think of including a message inside cmake (like behind "Added x tests").
Also, there is some value in some real, minimal tests to check that we can run ASPECT (configuration is sane, libraries exist, mpirun works, etc.).

**Comment by bangerth** [comment]
I was more thinking of simply printing text from some custom target when you call "make test", unless if you specified that special flag when calling cmake that would enable setting up the real testsuite. This at least takes care of the "make test" case. People calling ctest themselves likely know what they are doing.

**Comment by tjhei** [comment]
I am not able to reliably overwrite "make test". I figured out a way to print information by supplying a CTestCustom.ctest file. Is this acceptable now?

I still need to change the default test...

**Comment by tjhei** [comment]
okay, ready for review.

**Comment by tjhei** [comment]
(before we merge I need to adjust the official tester to run all tests!)

**Comment by tjhei** [comment]
This is ready to be merged!

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -106,9 +113,10 @@ LIST(SORT _tests)
 FOREACH(_test ${_tests})
   GET_FILENAME_COMPONENT(_test ${_test} NAME_WE)
 
-  CHECK_LINEAR_ALGEBRA(${CMAKE_CURRENT_SOURCE_DIR}/${_test}.prm)
+  SET(_use_test ON)
+  SHOULD_ENABLE_TEST(${CMAKE_CURRENT_SOURCE_DIR}/${_test}.prm)
 
-  IF("${_use_test}" STREQUAL "1")
+  IF(_use_test)
```
Comment (bangerth) [inline tests/CMakeLists.txt]: Ugh, this is what PARENT_SCOPE does -- know about the name of a particular variable in a different scope? I sure hate people designing programming languages who have no idea how to do it...

File: tests/CTestCustom.ctest.in [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      "echo '--------------------------------------------------------------'"
+      "echo 'Completed running the minimal testsuite. Please check the test'"
+      "echo 'results above. To run the complete testsuite, reconfigure your'"
+      "echo 'project with the option ASPECT_RUN_ALL_TESTS=ON or by running'"
+      "echo '    make setup_tests'"
+      "echo 'Be aware that installing the 'numdiff' tools is strongly'"
+      "echo 'recommended before running the full testsuite. Even then, the'"
+      "echo 'output is likely different on your machine due to configuration'"
+      "echo 'differences, so do not expect all tests to succeed on your'"
+      "echo 'machine.'"
```
Comment (bangerth) [inline tests/CTestCustom.ctest.in]: yes, nice!

**Comment by bangerth** [comment]
I like it! Thanks!

**Pull request merged**